examples/experimental/dagster-dlift/dagster_dlift/cloud_instance.py

```diff
@@ -112,3 +116,26 @@ def get_dbt_models(self, environment_id: int) -> Sequence[Mapping[str, Any]]:
                 "endCursor"
             ]
         return models
+
+    def get_dbt_sources(self, environment_id: int) -> Sequence[Mapping[str, Any]]:
+        sources = []
+        page_size = 100
+        start_cursor = 0
+        while response := self.make_discovery_api_query(
+            GET_DBT_SOURCES_QUERY,
+            {"environmentId": environment_id, "first": page_size, "after": start_cursor},
+        ):
+            sources.extend(
+                [
+                    source["node"]
+                    for source in response["data"]["environment"]["definition"]["sources"]["edges"]
+                ]
+            )
+            if not response["data"]["environment"]["definition"]["sources"]["pageInfo"][
+                "hasNextPage"
```

Comment:
seems like this while loop + pagination etc could be pulled out into some shared method, similar to how you've deduplicated code in the mocking test stuff?

Reply:
Yea I buy it; but I also don't want to couple it too finely before we know exactly what we need from all the components

Reply:
Ended up finding something that I think feels pretty good.